client/python/gradio_client/serializing.py

```diff
@@ -11,6 +11,9 @@
 
 
 class Serializable(ABC):
+    def __init__(self, root_url: str | None = None):
```

Comment:
I think the code is simpler if this is passed in as a root_url and it makes sense since each serializable will only have one root_url during it's lifetime. It does make it a bit awkward since `root_url` is also a parameter of the method.